### PR TITLE
Tunnel Trap Deletion

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Placing a trap then building a tunnel now destroys the trap. You can no longer weed over traps.

## Why It's Good For The Game

a

## Changelog

:cl:
fix: Building a tunnel over a resin hole now destroys the resin hole.
balance: You can no longer place a resin node over resin hole. And you can no longer place resin holes on resin nodes.
/:cl:

### CODE REVIEW

**Comment by DylonBR** [comment]
PIG nerf

File: code/modules/cm_aliens/structures/tunnel.dm [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
 		hive.tunnels += src
 
+	var/obj/effect/alien/resin/trap/resin_trap = locate() in L
+	if(resin_trap) qdel(resin_trap)
```
Comment (stanalbatross) [inline code/modules/cm_aliens/structures/tunnel.dm:45]:
```suggestion
	if(resin_trap) 
		qdel(resin_trap)
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]
we love readable code

**Review by stanalbatross — APPROVED** [review]

**Comment by SarahJohnson489** [comment]
Feedback:

Weed a room
Place a resin hole by the door
Destroy the hole and the weed tile under it leaving a 1x1 hole
Place a resin node to replace the weed
Attempt to place a resin hole

"You can't place a resin hole on a resin node!" "You can't place a resin hole on a resin node!"
"You can't place a resin hole on a resin node!" "You can't place a resin hole on a resin node!"
"You can't place a resin hole on a resin node!" "You can't place a resin hole on a resin node!"
"You can't place a resin hole on a resin node!" "You can't place a resin hole on a resin node!"


Grenades, CAS, wildin marine knives, c4, flames, a massive amount of sources cause non-orthagonal weed breakage while now results in denying burrowers tiles to place resin holes because of the change. Which is actually really kinda important given the extremely positional nature of Resin Hole placement. Needs a revert quickly.

**Comment by DeafeningAAA** [comment]
I remember talkin jeser out of adding this exact thing 2 years ago
The issue this change causes can be summed up like this:
>Random drone weeds an area
>Places weed nodes in a bunch of trap-friendly spots
>Someone else, playing burrower, can no longer access said trap spots, until marines destroy the nodes

Just revert. Placing traps on top of nodes was never a problem to begin with.

**Pull request merged**